Source/JavaScriptCore/dfg/DFGArrayMode.cpp

```diff
@@ -461,7 +470,17 @@ bool ArrayMode::alreadyChecked(Graph& graph, Node* node, const AbstractValue& va
 
     // Array::OriginalNonArray can be shown when the value is a TypedArray with original structure.
     // But here, we already filtered TypedArrays. So, just handle it like a NonArray.
-    case Array::OriginalNonArray:
+    case Array::OriginalNonArray: {
+        if (type() == Array::AlwaysSlowPutContiguous) {
+            if (!value.m_structure.isFinite())
+                return false;
+            JSGlobalObject* globalObject = graph.globalObjectFor(node->origin.semantic);
+            return value.m_structure.toStructureSet().isSubsetOf(globalObject->originalAlwaysSlowPutContiguousStructureSet());
+        }
```

Comment:
Should we call the other helper here?

Reply:
@saambarati Could please expand on "the other helper" here? Thanks.

Reply:
Not sure what I was thinking. Let's ignore this.